Potential Breaking Change in `resolveConfig`

It seems that 2.0 may have introduced an unintended breaking change in the `resolveConfig` function. This function is used in the Prettier VS Code extension here: https://github.com/prettier/prettier-vscode/blob/master/src/ConfigResolver.ts#L99

Since upgrading to 2.0 we are now seeing errors when trying to resolve configs:

```
Error: Cannot find module 'prettier-config-greenelab' from 'c:\Users\Vincent\Desktop\adage-frontend'
	at Function.Module._resolveFilename (internal/modules/cjs/loader.js:717:15)
	at Function.n.resolve (c:\Program Files\Microsoft VS Code\resources\app\out\vs\loader.js:15:584)
	at Object.transform (c:\Users\Vincent\.vscode\extensions\esbenp.prettier-vscode-4.0.0\node_modules\prettier\index.js:23283:40)
	at run (c:\Users\Vincent\.vscode\extensions\esbenp.prettier-vscode-4.0.0\node_modules\prettier\third-party.js:11472:51)
	at async cacheWrapper (c:\Users\Vincent\.vscode\extensions\esbenp.prettier-vscode-4.0.0\node_modules\prettier\third-party.js:11300:20)
	at async cacheWrapper (c:\Users\Vincent\.vscode\extensions\esbenp.prettier-vscode-4.0.0\node_modules\prettier\third-party.js:11300:20)
	at async cacheWrapper (c:\Users\Vincent\.vscode\extensions\esbenp.prettier-vscode-4.0.0\node_modules\prettier\third-party.js:11300:20)
	at async cacheWrapper (c:\Users\Vincent\.vscode\extensions\esbenp.prettier-vscode-4.0.0\node_modules\prettier\third-party.js:11300:20)
	at async cacheWrapper (c:\Users\Vincent\.vscode\extensions\esbenp.prettier-vscode-4.0.0\node_modules\prettier\third-party.js:11300:20)
	at async Explorer.search (c:\Users\Vincent\.vscode\extensions\esbenp.prettier-vscode-4.0.0\node_modules\prettier\third-party.js:11457:22)
	at async Promise.all (index 0)
```
 
For repro steps and more details see: https://github.com/prettier/prettier-vscode/issues/1289

One thing to note, is that everything behaves as expected from the CLI which is odd.

--- a/src/common/load-plugins.js
+++ b/src/common/load-plugins.js
@@ -8,6 +8,7 @@ const path = require("path");
 const thirdParty = require("./third-party");
 const internalPlugins = require("./internal-plugins");
 const mem = require("mem");
+const resolve = require("./resolve");
 
 const memoizedLoad = mem(load, { cacheKey: JSON.stringify });
 const memoizedSearch = mem(findPluginsInNodeModules);
@@ -42,14 +43,10 @@ function load(plugins, pluginSearchDirs) {
       let requirePath;
       try {
         // try local files
-        requirePath = eval("require").resolve(
-          path.resolve(process.cwd(), pluginName)
-        );
+        requirePath = resolve(path.resolve(process.cwd(), pluginName));
       } catch (_) {
         // try node modules
-        requirePath = eval("require").resolve(pluginName, {
-          paths: [process.cwd()],
-        });
+        requirePath = resolve(pluginName, { paths: [process.cwd()] });
       }
 
       return {
@@ -85,9 +82,7 @@ function load(plugins, pluginSearchDirs) {
 
       return memoizedSearch(nodeModulesDir).map((pluginName) => ({
         name: pluginName,
-        requirePath: eval("require").resolve(pluginName, {
-          paths: [resolvedPluginSearchDir],
-        }),
+        requirePath: resolve(pluginName, { paths: [resolvedPluginSearchDir] }),
       }));
     })
     .reduce((a, b) => a.concat(b), []);

--- a/src/common/resolve.js
+++ b/src/common/resolve.js
@@ -1,0 +1,12 @@
+"use strict";
+
+let { resolve } = eval("require");
+
+// In the VS Code extension `require` is overridden and `require.resolve` doesn't support the 2nd argument.
+if (resolve.length === 1) {
+  const Module = eval("require")("module");
+  const createRequire = Module.createRequire || Module.createRequireFromPath;
+  resolve = createRequire(__dirname).resolve;
+}
+
+module.exports = resolve;

--- a/src/config/resolve-config.js
+++ b/src/config/resolve-config.js
@@ -7,6 +7,7 @@ const mem = require("mem");
 
 const resolveEditorConfig = require("./resolve-config-editorconfig");
 const loadToml = require("../utils/load-toml");
+const resolve = require("../common/resolve");
 
 const getExplorerMemoized = mem(
   (opts) => {
@@ -18,9 +19,7 @@ const getExplorerMemoized = mem(
           if (typeof result.config === "string") {
             const dir = path.dirname(result.filepath);
             try {
-              const modulePath = eval("require").resolve(result.config, {
-                paths: [dir],
-              });
+              const modulePath = resolve(result.config, { paths: [dir] });
               result.config = eval("require")(modulePath);
             } catch (error) {
               // Original message contains `__filename`, can't pass tests

